stages/test/test_skopeo.py

```diff
@@ -33,3 +43,84 @@ def test_schema_validation_skopeo(stage_schema, test_data, expected_err):
     else:
         assert res.valid is False
         testutil.assert_jsonschema_error_contains(res, expected_err, expected_num_errs=1)
+
+
+@pytest.mark.skipif(os.getuid() != 0, reason="needs root")
+@pytest.mark.skipif(not has_executable("podman"), reason="no podman executable")
+@pytest.mark.parametrize("dest_type,local_output_path", [
+    ("dir", "/tmp/test-output-skopeo-dir"),
+    ("oci-archive", "/tmp/test-output-skopeo.tar"),
+    ("oci", "/tmp/test-output-skopeo-dir"),
+])
+def test_skopeo_copy(tmp_path, stage_module, dest_type, local_output_path):
+    with make_container(tmp_path, {"file1": "file1 from final layer"}) as cont_tag:
+        # export for the container-deploy stage
+        fake_container_dst = tmp_path / "fake-container"
+        subprocess.check_call([
+            "podman", "save",
+            "--format=oci-archive",
+            f"--output={fake_container_dst}",
+            cont_tag,
+        ])
+
+    inputs = {
+        "images": {
+            # seems to be unused with fake_container_path?
+            "path": fake_container_dst,
+            "data": {
+                "archives": {
+                    fake_container_dst: {
+                        "format": "oci-archive",
+                        "name": cont_tag,
+                    },
+                },
+            },
+        },
+    }
```

Comment:
This input is _technically_ invalid but it's working because the container input parser is `join()`ing `fake_container_dst` from the `path` with `fake_container_dst` from the `archives` keys.  The `path` is meant to be the path to the osbuild store for container archives, so something like `<osbuild-store>/sources/org.osbuild.containers/` and the key below `archives` is meant to be the checksum of the container, so the paths get joined into `<osbuild-store>/sources/org.osbuild.containers/sha256:aaaaaaa...`.

The `join()` works out here because doing an `os.path.join()` with two abs paths drops the first one and just returns the second as is.

Reply:
it's actually a copy'n'paste from https://github.com/osbuild/osbuild/blob/main/stages/test/test_container_deploy.py#L35
😟 I'll align with @mvo5 to follow up on this

Reply:
Yeah, I can take care of this, sorry that this was a bad example.

Reply:
I put the fix in https://github.com/osbuild/osbuild/pull/1761 thanks again @schuellerf and @achilleas-k for noticing!